### PR TITLE
Add connection between Card and board

### PR DESCRIPTION
**Pull request by psmishenko**
https://trello.com/c/Lnat2C3I/36-add-connection-between-card-and-board
Снимок экрана от 2022-06-02 10-40-29
Снимок экрана от 2022-06-02 10-40-44

### CODE REVIEW

File: app/controllers/cards_controller.rb [hunk 1]
```diff
@@ -38,6 +49,10 @@ def card_params
   end
 
   def current_card
-    @card = Card.find(params[:id])
+    @card = @board.cards.find(params[:id])
+  end
+
+  def current_board
+    @board = Board.find(params[:board_id])
   end
```
Comment (AnastasiyaKireyeva) [inline app/controllers/cards_controller.rb]: let's rename the methods ( find_card & find_board )

File: app/models/card.rb [hunk 2]
```diff
@@ -2,4 +2,5 @@
 
 class Card < ApplicationRecord
   belongs_to :board
+  validates :question, :answer, presence: true
```
Comment (B0RlS) [inline app/models/card.rb]: Could you please add length validation?
And add specs please

File: app/controllers/cards_controller.rb [hunk 3]
```diff
@@ -2,32 +2,43 @@
 
 class CardsController < ApplicationController
   before_action :authenticate_user!
+  before_action :current_board
   before_action :current_card, except: %i[index new create]
 
   def index
-    @cards = Card.all
+    @cards = @board.cards
```
Comment (AnastasiyaKireyeva) [inline app/controllers/cards_controller.rb]: can we show the cards in a descending order?

File: app/controllers/cards_controller.rb [hunk 4]
```diff
@@ -2,32 +2,43 @@
...
+      end
+    end
   end
 
   def update
     @card.update(card_params)
-
-    redirect_to card_path(@card)
+    respond_to do |format|
+      if @card.update(card_params)
```
Comment (AnastasiyaKireyeva) [inline app/controllers/cards_controller.rb]: if @card.update(card_params)
      respond_to do |format|

File: app/views/cards/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<%= form_with model: [@board,card] do |form| %>
```
Comment (AnastasiyaKireyeva) [inline app/views/cards/_form.html.erb:1]: I suppose, we should consider styles in views too ( I mean at least indents )

File: app/controllers/cards_controller.rb (hunk 4 above)
Comment (DashaSledopit) [reply app/controllers/cards_controller.rb:30]: the same can be done in the create action

**Pull request merged**